bip-taproot.mediawiki

```diff
@@ -62,22 +62,22 @@ In the text below, ''hash<sub>tag</sub>(m)'' is a shorthand for ''SHA256(SHA256(
 * If there is exactly one element left in the witness stack, key path spending is used:
-** The single witness stack element is interpreted as the signature and must be valid (see the next section) for the public key ''Q'' and taproot transaction digest (to be defined hereinafter) as message. Fail if it is not. Otherwise pass.
+** The single witness stack element is interpreted as the signature and must be valid (see the next section) for public key ''q'' and taproot transaction digest (to be defined hereinafter) as message. Fail if it is not. Otherwise pass.
 * If there are at least two witness elements left, script path spending is used:
 ** Call the second-to-last stack element ''s'', the script.
 ** The last stack element is called the control block ''c'', and must have length ''33 + 32m'', for a value of ''m'' that is an integer between 0 and 32, inclusive. Fail if it does not have such a length.
-** Let ''P = point(byte(2 + (c[0] & 1)) || c[1:33])''<ref>'''What is the purpose of the first byte of the control block?''' The first byte of the control block has three distinct functions:
+** Let ''P = point(byte(2 + (c[0] & 1)) || c[1:33])''<ref>'''What is the purpose of the first byte of the control block?''' The first byte of the control block has four distinct functions:
 * The low bit is used to denote the oddness of the Y coordinate of the ''P'' point.
+* The second bit is used to denote the oddness of the Y coordinate of the ''Q'' point.<ref>'''Why is the Y coordinate of the output public key required in a script path spend?''' Because ''Q'' is constructed by adding the taproot tweak to the internal public key ''P'', it cannot easily be ensured that ''Q'' has an even Y coordinate. Trying out multiple internal keys until there's such a ''Q'' is possible but undesirable and unnecessary since signalling the oddness of the Y coordinate only consumes an unused bit.</ref>
```

Comment:
I wonder if it wouldn't make for an easier explanation to treat the ''Q'' point as always even, and have the second bit used to denote whether the formula is negated?

Reply:
(That would make the later test be `dQ = P + int(t)G` where `d=1` if `c[0]&2==0` and `d=-1` otherwise)

Reply:
I think so if we only have 32 byte keys